packages/compass-crud/src/components/json-editor.jsx

```diff
@@ -94,87 +65,104 @@ class EditableJson extends React.Component {
    * Fold up all nested values when loading editors.
    */
   componentDidMount() {
+    this.subscribeToDocumentEvents(this.props.doc);
     this.editor.getSession().foldAll(2);
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.doc !== this.props.doc) {
+      this.unsubscribeFromDocumentEvents(prevProps.doc);
+      this.subscribeToDocumentEvents(this.props.doc);
+      if (this.state.editing || this.state.deleting) {
+        // eslint-disable-next-line react/no-did-update-set-state
+        this.setState({ editing: false, deleting: false });
+      }
+    }
+  }
+
   /**
    * Subscribe to the update store on mount.
    *
    * Fold all documents on update as well, since we might get fresh documents
    * from the query bar.
    */
-  componentDidUpdate(prevProps) {
-    if (!this.state.editing) {
-      this.editor.getSession().foldAll(2);
-    }
-    if (this.state.editing && this.props.updateError) {
-      this.handleUpdateError();
-    } else if (this.state.deleting && this.props.updateSuccess) {
-      this.handleRemoveSuccess();
-    } else if (this.state.editing && this.props.updateSuccess) {
-      this.handleUpdateSuccess();
-    } else if (this.props.doc !== prevProps.doc) {
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
Looks like there are two `componentDidUpdate` methods now, should we consolidate them?

Reply:
Oh lol, I swear this react components methods sort order eslint rule makes me do the silliest things. Thanks for spotting, these should indeed just be one now